bob-backend/src/pearl/holder.rs

```diff
@@ -187,11 +187,11 @@ impl Holder {
 
     // @TODO remove redundant return result
     #[allow(clippy::cast_possible_truncation)]
-    async fn write_disk(storage: PearlStorage, key: Key, data: BobData) -> BackendResult<()> {
+    async fn write_disk(storage: PearlStorage, key: Key, data: &BobData) -> BackendResult<()> {
         counter!(PEARL_PUT_COUNTER, 1);
         let data_size = Self::calc_data_size(&data);
         let timer = Instant::now();
-        let res = storage.write(key, Data::from(data).to_vec()).await;
+        let res = storage.write(key, data.inner().to_vec()).await;
```

Comment:
Can this data copying be avoided?

Reply:
Only if we change pearl's interface to accept value by ref
https://github.com/qoollo/pearl/blob/master/src/storage/core.rs#L249

Reply:
Ok, I created a separate issue for that particular task: https://github.com/qoollo/pearl/issues/191